cylc/flow/etc/cylc

```diff
@@ -17,95 +17,71 @@
-# or conda environment, and use CYLC_ENV_NAME to select it.
-#
-#   $ CYLC_ENV_NAME=cylc-violet-capybara cylc version --long
-#   > 8.0.0 (/home/user/miniconda3/envs/cylc-violet-capybara)
-#
-# INSTALLING LEGACY cylc 7 RELEASE TARBALLS BY HAND
-#--------------------------------------------------
-# cylc-flow release tarballs now unpack to (e.g.) cylc-flow-7.9.1. To work with
-# this wrapper the directory should be renamed to "cylc-7.9.1". Then follow
-# version-specific installation instructions. Running "make" should create a
-# file called VERSION that contains just the version string (e.g.) "7.9.1".
-#
-# INSTRUCTIONS FOR USERS
-#-----------------------
-# + Set CYLC_HOME_ROOT_ALT to point local conda releases, e.g.:
-#     $ export CYLC_HOME_ROOT_ALT=$HOME/miniconda3/envs
-# + Set CYLC_VERSION e.g. "8.0.0" to select a specific version in the root
-#     locations. CYLC_VERSION is propagated to task jobs by the scheduler; to
-#     avoid overiding this you should only default to your version:
-#       $ export CYLC_VERSION=${CYLC_VERSION:-8.0.0}
-#   - Do not explicitly select the default "cylc" symlink as a version
-# + Set CYLC_HOME to select a specific Cylc 8 venv or plain Cylc 7 directory
-#   outside of the ROOT locations
-# + Set CYLC_ENV_NAME to a select a specific arbitrarily name Cylc venv (directly,
-#   or under the ROOT locations).
-# + These settings (e.g. in .bashrc) must be replicated on job hosts too.
+# CYLC_VERSION - can be set by users (e.g. in .bash_profile) in order to select a
+# specific Cylc environment. If set this wrapper will look for an installed
+# environment called cylc-$CYLC_VERSION in the ROOT locations (if not set it will
+# look for an installed environment called cylc).
+# e.g.    export CYLC_VERSION=8.0.0-1
+#
+# CYLC_HOME_ROOT - location of installed Cylc environments.
+# Usually defined in this wrapper script.
+#
+# CYLC_HOME_ROOT_ALT - alternate location of installed Cylc environments.
+# Can be set by users for use with their own Cylc environments.
+# If used it must be set on workflow and job hosts, e.g. in .bash_profile.
+# e.g.    export CYLC_HOME_ROOT_ALT=$HOME/miniconda3/envs
+#
+# CYLC_ENV_NAME - gets set to the basename of the selected environment.
+# If set this defines the environment name rather than CYLC_VERSION.
+# The scheduler sets CYLC_ENV_NAME for all remote commands to ensure the same
+# environment is used on all platforms.
+# Users should not set CYLC_ENV_NAME (use CYLC_VERSION).
+#
+# CYLC_HOME - full path to the Cylc environment.
+# Can be set by users in order to use an environment outside of the ROOT
+# locations. However, it is not passed by the scheduler to remote platforms so
+# use of CYLC_VERSION & CYLC_HOME_ROOT_ALT is preferred.
+#
+# INSTALLING ENVIRONMENTS
+# -----------------------
+#
+# Releases should be installed into environments in the ROOT location, named
+# cylc-$CYLC_VERSION. We recommend using the cylc-flow version plus an additional
```

Comment:
A small change would be handy at NIWA, to avoid having to edit some code down in the guts of the wrapper:
```
CYLC_ENV_NAME_PREFIX="cylc-"   # editable prefix
# ...
${CYLC_ENV_NAME_PREFIX}${CYLC_VERSION}
```
Our HPC admins use this convention: `.../share/cylc/8.0rc1` etc.

Reply:
(However, not a biggie)

Reply:
I like this idea. However, I think it also means making the name of the default version configurable (also like the idea of this). Do you mind if we do this in a separate PR?

Reply:
See https://github.com/cylc/cylc-flow/issues/4555#issuecomment-1066618751